flytepropeller/pkg/controller/nodes/array/handler.go

```diff
@@ -547,6 +551,18 @@ func (a *arrayNodeHandler) Handle(ctx context.Context, nCtx interfaces.NodeExecu
 			return handler.UnknownTransition, fmt.Errorf("worker error(s) encountered: %s", workerErrorCollector.Summary(events.MaxErrorMessageLength))
 		}
 
+		// only offload literal if canOffloadLiteral is true which currently checks if the sdk is compatible to use with this backend change.
+		if canOffloadLiteral {
+			for outputLiteralKey, outputLiteral := range outputLiterals {
+				// if the size of the output Literal is > threshold then we write the literal to the offloaded store and populate the literal with its zero value and update the offloaded url
+				// use the OffloadLargeLiteralKey to create  {OffloadLargeLiteralKey}_offloaded_metadata.pb file in the datastore.
+				// Update the url in the outputLiteral with the offloaded url and also update the size of the literal.
+				offloadedOutputFile := v1alpha1.GetOutputsLiteralMetadataFile(outputLiteralKey, nCtx.NodeStatus().GetOutputDir())
+				if err := common.OffloadLargeLiteral(ctx, nCtx.DataStore(), offloadedOutputFile, outputLiteral, a.literalOffloadingConfig); err != nil {
```

Comment:
This is offloading data from individual subNode executions right? Below this is where we actually write the aggregated data (~570). IIUC that is where we actually want to offload. For example, each subNode outputs 50M, which is under the offloading configuration threshold, but if there are 1000 subnodes that exceeds the threshold.